### PR TITLE
SDC Expansion Idempotency Fix

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
SDC Expansion Idempotency Fix

image

### CODE REVIEW

**Comment by Krishnan-Priyanshu** [comment]
did we removed the plan modifier?

File: powerflex/sdc_expansion_resource_schema.go [hunk 1]
```diff
@@ -121,9 +121,6 @@ var csvSchema schema.SetNestedAttribute = schema.SetNestedAttribute{
 				Description:         "Whether this works as MDM or Tie Breaker,The acceptable value is `Primary`, `Secondary`, `TB`, `Standby` or blank. Default value is blank",
 				Optional:            true,
 				MarkdownDescription: "Whether this works as MDM or Tie Breaker,The acceptable value is `Primary`, `Secondary`, `TB`, `Standby` or blank. Default value is blank",
-				PlanModifiers: []planmodifier.String{
-					stringDefault(" "),
-				},
 			},
```
Comment (rounak-adhikary) [inline powerflex/sdc_expansion_resource_schema.go:124]: If we have idempotency concerns, shouldnt we use the UseStateForUnknown (Ref: https://github.com/dell/terraform-provider-powerflex/blob/main/powerflex/sdc_volumes_mapping_resource.go#L132) planmodifier? Have we tested if this change is solving idempotency?

Comment (Krunal-Thakkar) [reply powerflex/sdc_expansion_resource_schema.go:124]: Yeah, it is working. First I was required value, but now that is handled internally.

**Comment by Krunal-Thakkar** [comment]
> did we removed the plan modifier?

yes because the default value affecting in idempotency checks, so handled internally.

**Review by shenda1 — APPROVED** [review]

**Pull request merged**